enterprise/api_client/enterprise.py

```diff
@@ -56,6 +56,8 @@ def get_enterprise_course_runs(self, enterprise_customer):
                 self.ENTERPRISE_CUSTOMER_CATALOGS_ENDPOINT,
                 resource_id=str(enterprise_customer_catalog.uuid),
                 traverse_pagination=True,
+                # we need to fetch data in large chunks so that we do not hit api limit.
+                querystring={'page_size': 1000},
```

Comment:
Isn't [this](https://github.com/edx/edx-enterprise/blob/54003d767ae81b42de1357b464c2c86ea64490dc/enterprise/models.py#L921) going to prevent us from traversing pagination?

Reply:
Actually we are traversing pagination [here](https://github.com/edx/edx-enterprise/blob/master/enterprise/api_client/enterprise.py#L126).

[This](https://github.com/edx/edx-enterprise/blob/54003d767ae81b42de1357b464c2c86ea64490dc/enterprise/models.py#L921) part of the code causes enterprise API to return paginated response and [this](https://github.com/edx/edx-enterprise/blob/master/enterprise/api_client/enterprise.py#L126) part makes sure to traverse pagination of enterprise API and return the all the content.

Reply:
Got it. Makes sense. Thanks.

Can we move the addition of the `page_size` query parameter to the `utils.traverse_pagination function`? If we are traversing pagination, seems like we should always be doing that in the largest chunks possible.

Reply:
`utils.traverse_pagination` [follows pagination based on `next`](https://github.com/edx/edx-enterprise/blob/master/enterprise/utils.py#L541) url in the response dict which already contain `page`, `page_size`, `offset`, `limit` etc. parameters. 

So, I am not sure if it is possible to add `page_size` argument to `utils.traverse_pagination`. The reason is that `utils.traverse_pagination` is designed to be independent of pagination style (e.g. page, page_size vs limit, offset).

Reply:
Makes sense. Thanks for clarifying.